spring-cloud-task-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -153,3 +153,184 @@ public class TimestampTaskApplication {
 
 After doing so, you can simply add the additional configuration for the extra features of your application.
 
+=== Developing custom task applications and generating artifacts using the existing tooling
+
+In this section, we will explain how to develop a custom task application and then generate
+maven and docker artifacts for it using the existing tooling provided by the
+spring cloud task app starter infrastructure. For explanation purposes, we will assume that we are creating a new
+task application for a technology named foobar.
+
+* Create a new top level module named spring-cloud-starter-task-foobar
+
+Please follow the instructions above for designing a proper Spring Cloud Task starter. You may also look into the existing
+starters for how to structure a new one. Ensure that you name the main `@Configuration` class of your starter as
+`FoobarTaskConfiguration` as this is the default convention used by the app generation later. Please also make sure
+that the package for this `@Configuration` is `org.springfamework.cloud.task.app.foobar`. If you have a different
+class/package name, see below for how to override that in the app generator. The functionality name for which you create
+a starter can be a hyphenated stream of strings such as in `spark-client` which can be found in the
+module `spring-cloud-starter-task-spark-client`.
+
+The starters in `spring-cloud-task-app-starters` are slightly different from the other starters in spring-boot and
+spring-cloud in that here we don't provide a way to auto configure any configuration through spring factories mechanism.
+Rather, we delegate this responsibility to the maven plugin that is generating the binder based apps. Therefore, you don't
+have to provide a spring.factories file that lists all your configuration classes.
+
+* Add the new foobar task starter module to the root pom of the repository
+
+* At the root of the repository build and install the new module into your local maven cache:
+
+`./mvnw clean install -pl :spring-cloud-starter-task-foobar`
+
+* You need to add the new starter dependency to the `spring-cloud-task-app-dependencies` bill of material (BOM) in the
+dependecy management section. For example,
+
+[source,xml]
+----
+<dependencyManagement>
+...
+...
+    <dependency>
+        <groupId>org.springframework.cloud.task.app</groupId>
+        <artifactId>spring-cloud-starter-task-foobar</artifactId>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
+    </dependency>
+...
+...
+----
+
+* Build and install the newly updated bom:
+
+`./mvnw clean install -pl :spring-cloud-task-app-dependencies`
+
+* At this point, you are ready to generate the spring boot app for foobar task. Go to the
+`spring-cloud-task-app-generator` module and start editing as below.
+
+If you have a very basic task starter, all you have to do is to add the following to the list of apps need to be
```

Comment:
This sentence is confusing.  What about `The minimal configuration needed to generate the app is to add <foobar-task/> to plugin configuration in spring-cloud-task-app-generator/pom.xml.  There are other plugin options that customize the generated applications which are described in the plugin documentation (https://github.com/spring-cloud/spring-cloud-stream-app-maven-plugin)